src/Provider/GoogleMaps/GoogleMaps.php

```diff
@@ -264,9 +264,42 @@ private function fetchUrl(string $url, string $locale = null, int $limit, string
             if (isset($result->formatted_address)) {
                 $address = $address->withFormattedAddress($result->formatted_address);
             }
+            if ($builder->hasValue('street_address')) {
+                $address = $address->withStreetAddress($builder->getValue('street_address'));
+            }
```

Comment:
Yes of course for all of these parameters. 

The result will be exactly the same and the code will get easier to read and far less complex (nPath)